--- a/src/core/ozone/gl_context_qt.cpp
+++ b/src/core/ozone/gl_context_qt.cpp
@@ -60,6 +60,7 @@ namespace {
 inline void *resourceForContext(const QByteArray &resource)
 {
 #ifndef QT_NO_OPENGL
+    LOG(INFO) << "calling resourceForContext: " << QString(resource).toStdString(); 
     QOpenGLContext *shareContext = qt_gl_global_share_context();
     if (!shareContext) {
         qFatal("QWebEngine: OpenGL resource sharing is not set up in QtQuick. Please make sure to call QtWebEngine::initialize() in your main() function.");
