--- a/src/buildtools/config/linux.pri
+++ b/src/buildtools/config/linux.pri
@@ -91,16 +91,7 @@ contains(QT_ARCH, "arm") {
         }
     }
 
-    qtConfig(webengine-arm-thumb) {
         gn_args += arm_use_thumb=true # this adds -mthumb
-    } else {
-        gn_args += arm_use_thumb=false
-        !qtConfig(webengine-system-ffmpeg) {
-             # Fixme QTBUG-71772
-             gn_args += media_use_ffmpeg=false
-             gn_args += use_webaudio_ffmpeg=false
-        }
-    }
 }
 
 contains(QT_ARCH, "mips") {
