--- a/src/core/net/network_delegate_qt.cpp
+++ b/src/core/net/network_delegate_qt.cpp
@@ -299,7 +299,8 @@ bool NetworkDelegateQt::OnCanQueueReport
 
 void NetworkDelegateQt::OnCanSendReportingReports(std::set<url::Origin> origins, base::OnceCallback<void(std::set<url::Origin>)> result_callback) const
 {
-    std::move(result_callback).Run(std::set<url::Origin>());
+    origins.clear();
+    std::move(result_callback).Run(std::move(origins));
 }

