--- a/src/core/ozone/gl_ozone_egl_qt.cpp
+++ b/src/core/ozone/gl_ozone_egl_qt.cpp
@@ -100,11 +100,11 @@ bool GLOzoneEGLQt::LoadGLES2Bindings(gl:
 
     if (!get_proc_address) {
         LOG(ERROR) << "eglGetProcAddress not found.";
-        base::UnloadNativeLibrary(eglgles2Library);
-        return false;
+        //base::UnloadNativeLibrary(eglgles2Library);
+        //return false;
     }
 
-    gl::SetGLGetProcAddressProc(get_proc_address);
+    //gl::SetGLGetProcAddressProc(get_proc_address);
     gl::AddGLNativeLibrary(eglgles2Library);
     return true;
 }
