--- a/src/core/ozone/gl_ozone_egl_qt.cpp
+++ b/src/core/ozone/gl_ozone_egl_qt.cpp
@@ -78,7 +78,7 @@ base::NativeLibrary LoadLibrary(const ba
 
 bool GLOzoneEGLQt::LoadGLES2Bindings(gl::GLImplementation /*implementation*/)
 {
-    base::NativeLibrary eglgles2Library = dlopen(NULL, RTLD_LAZY);
+    base::NativeLibrary eglgles2Library = dlopen("libEGL.so.1", RTLD_LAZY);
     if (!eglgles2Library) {
         LOG(ERROR) << "Failed to open EGL/GLES2 context " << dlerror();
         return false;
