--- a/src/core/ozone/gl_ozone_egl_qt.cpp
+++ b/src/core/ozone/gl_ozone_egl_qt.cpp
@@ -99,6 +99,11 @@ bool GLOzoneEGLQt::LoadGLES2Bindings(gl:
 #endif
 
     if (!get_proc_address) {
+       LOG(ERROR) << "workaround (static)";
+       get_proc_address = &eglGetProcAddress;
+    }
+
+    if (!get_proc_address) {
         LOG(ERROR) << "eglGetProcAddress not found.";
         base::UnloadNativeLibrary(eglgles2Library);
         return false;
