--- a/src/webengine/api/qquickwebengineview_p.h
+++ b/src/webengine/api/qquickwebengineview_p.h
@@ -510,6 +510,8 @@
 
     LifecycleState recommendedState() const;
 
+    void setHeight(qreal h) { QQuickItem::setHeight(qRound(h)); }
+
 public Q_SLOTS:
     void runJavaScript(const QString&, const QJSValue & = QJSValue());
     Q_REVISION(3) void runJavaScript(const QString&, quint32 worldId, const QJSValue & = QJSValue());
