Description: print compiler commands when bootstrapping gn
Author: Dmitry Shachnev <mitya57@debian.org>
Forwarded: no
Last-Update: 2017-07-20

--- a/src/buildtools/gn.pro
+++ b/src/buildtools/gn.pro
@@ -25,7 +25,7 @@ build_pass|!debug_and_release {
               gn_args = $$replace(gn_args, "use_incremental_linking=true ", "")
             }
 
-            gn_configure = $$system_quote($$gn_bootstrap) --shadow --gn-gen-args=$$gn_args $$ninja_path
+            gn_configure = $$system_quote($$gn_bootstrap) --verbose --shadow --gn-gen-args=$$gn_args $$ninja_path
             !system("cd $$system_quote($$system_path($$dirname(out))) && $$pythonPathForSystem() $$gn_configure") {
                 error("GN build error!")
             }
